vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -122,33 +129,66 @@ def _action_outputs(self) -> dict[str, _IdProperty]:
-        try:
-            # First try to treat value as a column name. Unfortunately the customdata returned in the trigger does
+    def _get_value_from_trigger(self, value: str, trigger: dict[str, list[JsonValue]]) -> MultiValueType | None:
+        """Extract values from the trigger that represents clicked or selected Plotly graph points.
+
+        Priority:
+          1) If `value` matches a column name in self["custom_data"], take it as from customdata[index].
+          2) Otherwise treat `value` as a Box lookup (e.g. "x", "customdata[0]").
+
+        Notes:
+          - Allows camelCase and snake_case value keys interchangeably (camel_killer_box=True).
+          - Enables dot-style (e.g. value="customdata[0]") access to nested dict values (box_dots=True).
+          - Automatically creates missing keys as empty boxes instead of raising errors (default_box=True). This is done
+            to avoid exceptions when the `trigger` has a key with a dot in it.
+
+        Returns:
+          - list of values (one per point) or None if no points selected (signals reset).
+
+        Raises:
+          - ValueError if `value` can't be found.
+        """
+        # Returning None signals a reset of control to its original value.
+        if not (trigger or {}).get("points"):
+            return None
+
+        trigger_box = BoxList(trigger["points"], camel_killer_box=True, box_dots=True, default_box=True)
+
+        lookup_keys = []
+        with suppress(KeyError, ValueError):
+            # First try to treat value as a column name. Unfortunately, the customdata returned in the trigger does
             # not contain column names (it's just a list) so we must look it up in the called function's `custom_data`
             # to find its numerical index in this list. This only works if a custom_data was provided in the graph
             # function call.
             index = self["custom_data"].index(value)
-            return trigger_box["customdata"][index]
-        except (KeyError, ValueError):
-            try:
-                # Treat the value as a box lookup string, as in
-                # https://github.com/cdgriffith/Box/wiki/Types-of-Boxes#box-dots. This works for e.g. value="x" and
-                # value="customdata[0]".
-                return trigger_box[value]
-            except (KeyError, TypeError):
-                raise ValueError(
-                    f"Couldn't find value `{value}` in trigger for `set_control` action. "
-                    f"This action was added to the Graph model with ID `{self.id}`. "
-                    "If you expected the value to come from custom data, add it in the figure's custom_data signature."
-                )
+            lookup_keys.append(f"customdata[{index}]")
+
+        # Treat the value as a box lookup string, as in https://github.com/cdgriffith/Box/wiki/Types-of-Boxes#box-dots
+        # This works for e.g. value="x" or value="customdata[0]".
+        lookup_keys.append(value)
+
```

Comment:
I don't quite understand the logic here. Why could there be multiple lookup keys? Don't we want to treat it as we either have `value=<column_name>` or `value=<box-lookup-string>`?

I was expecting something a bit more like this:

```suggestion
try:
    # Try value=<column_name>
    index = self["custom_data"].index(value)
    lookup = f"customdata[{index}]"
except (KeyError, ValueError):
    # If that fails, use value=<box-lookup-string>
    lookup = value

try:
    return [point[lookup] for point in trigger_box]
except (KeyError, TypeError):
    raise ValuError("...")
```

Reply:
I don't know why, but I thought that I have to visit the `lookup = value` even when the `lookup = f"customdata[{index}]"` fails. So, I decided to "refactor it like this" 😄. I'm fixing it..